### ISSUE TITLE
Array to string conversion in AssetUsageIndex.php line 202

### ISSUE THREAD

**Issue by kdambekalns** [bwaidelich/Wwwision.AssetUsage#2]
I got this (thounsands of times) when running `./flow assetusage:updateIndex`:

`Error while processing node "f8971542-ac36-45a3-80e1-2fe30c3c565a": Failed to update usages for node "f8971542-ac36-45a3-80e1-2fe30c3c565a": Failed to extract asset ids from property "imageDesk": Warning: Array to string conversion in /var/www/app/Data/Temporary/Development/SubContextDocker/Cache/Code/Flow_Object_Classes/Wwwision_AssetUsage_AssetUsageIndex.php line 202`

I'll try to find out more…

**Comment by bwaidelich** [comment]
Thanks for reporting.
This is the line that produces the error: https://github.com/bwaidelich/Wwwision.AssetUsage/blob/a8aa47583dc3f101117157454e4c400867d50e97/Classes/AssetUsageIndex.php#L202

I could imagine that `$value` is an array in this case (which it should not be according to the type – unless it's an object (without `__toString()` and not implementing `ResourceBasedInterface`) that is serialized as an array).
If you can share the contents (anonymized) of the property in question, I might be able to provide a fix

**Comment by bwaidelich** [comment]
Can you test whether chaning the `$type === 'string'` to `is_string($value)` in that line fixes the issue for you?

**Comment by kdambekalns** [comment]
Here we go, `$type` and `$value` at that point are something like this:

```
string(6) "string"
array(2) {
  ["__flow_object_type"]=>
  string(29) "Neos\Media\Domain\Model\Image"
  ["__identifier"]=>
  string(36) "92260671-2f1f-4a17-a03c-85eeb762c6f8"
}
```

**Comment by kdambekalns** [comment]
This seems to be caused by a property that is not defined correctly?!

**Comment by kdambekalns** [comment]
Yep, properties used to store `Neos\Media\Domain\Model\ImageInterface` not being defined, aside from a measly:

```yaml
imageSmall:
  ui:
    label: i18n
```

Now I need to find out if the properties need to be fixed or the database contains leftover data… and then thnk about a way to fix that warning in the best way.

**Comment by kdambekalns** [comment]
It seems the database has leftover data from a change to the way images are handled in the project. 🙄

And I think changing the check from `$type === 'string'` to `is_string($value)` is actually fine…

**Comment by bwaidelich** [comment]
~I thought about this again and I think I deliberately didn't check the type of the (serialized) value here, because that could slow down the process quite a bit.~
~In your case you're just missing a proper type in the node type schema it seems.~
~So +1 for improving the error handling, but I'm a bit skeptical about ignoring the declared type~

ignore this, your PR makes sense!

**Comment by bwaidelich** [comment]
@kdambekalns I tagged the fixed version as [1.1.0](https://github.com/bwaidelich/Wwwision.AssetUsage/releases/tag/1.1.0) thanks
